src/tests/D3MHub.t.sol

```diff
@@ -1638,298 +1201,266 @@ contract D3MHubTest is DssTest {
+        (uint256 ink, uint256 art) = vat.urns(ilk, address(pool));
         assertEq(ink, 50 * WAD);
         assertEq(art, 50 * WAD);
         uint256 prevDai = vat.dai(vow);
 
-        d3mHub.exec(ilk);
+        hub.exec(ilk);
 
-        (ink, art) = vat.urns(ilk, address(d3mTestPool));
+        (ink, art) = vat.urns(ilk, address(pool));
         assertEq(ink, 55 * WAD);
         assertEq(art, 55 * WAD);
         assertEq(vat.dai(vow), prevDai + 5 * RAD);
-        assertEq(testGem.balanceOf(address(d3mTestPool)), 60 * WAD);
+        assertEq(testGem.balanceOf(address(pool)), 60 * WAD);
     }
 
     function test_exec_fixInk_limited_under_debt_ceiling_something_to_withdraw() public {
         _windSystem();
         // interest is determined by the difference in gem balance to dai debt
         // by giving extra gems to the Join we simulate interest
-        _giveTokens(TokenLike(address(testGem)), 10 * WAD);
-        testGem.transfer(address(d3mTestPool), 10 * WAD); // Simulates 10 WAD of interest accumulated
-        assertEq(testGem.balanceOf(address(d3mTestPool)), 60 * WAD);
-        vm.store(
-            address(dai),
-            keccak256(abi.encode(address(testGem), uint256(2))),
-            bytes32(uint256(3 * WAD))
-        );
-        assertEq(d3mTestPool.maxWithdraw(), 3 * WAD);
+        address(testGem).setBalance(address(pool), 10 * WAD);
+        assertEq(pool.assetBalance(), 60 * WAD);
+        pool.decreaseIdleLiquidity(47 * WAD);
+        assertEq(pool.maxWithdraw(), 3 * WAD);
 
         vat.file(ilk, "line", 55 * RAD);
 
-        (uint256 ink, uint256 art) = vat.urns(ilk, address(d3mTestPool));
+        (uint256 ink, uint256 art) = vat.urns(ilk, address(pool));
         assertEq(ink, 50 * WAD);
         assertEq(art, 50 * WAD);
         uint256 prevDai = vat.dai(vow);
 
-        d3mHub.exec(ilk);
+        hub.exec(ilk);
 
-        (ink, art) = vat.urns(ilk, address(d3mTestPool));
+        (ink, art) = vat.urns(ilk, address(pool));
         assertEq(ink, 55 * WAD);
         assertEq(art, 55 * WAD);
         assertEq(vat.dai(vow), prevDai + 8 * RAD);
-        assertEq(testGem.balanceOf(address(d3mTestPool)), 57 * WAD);
+        assertEq(testGem.balanceOf(address(pool)), 57 * WAD);
     }
 
     function test_exec_fixInk_full_at_debt_ceiling() public {
         _windSystem();
         // interest is determined by the difference in gem balance to dai debt
         // by giving extra gems to the Join we simulate interest
-        _giveTokens(TokenLike(address(testGem)), 10 * WAD);
-        testGem.transfer(address(d3mTestPool), 10 * WAD); // Simulates 10 WAD of interest accumulated
-        assertEq(testGem.balanceOf(address(d3mTestPool)), 60 * WAD);
-        assertEq(d3mTestPool.maxWithdraw(), 50 * WAD);
+        address(testGem).setBalance(address(this), 10 * WAD);
+        testGem.transfer(address(pool), 10 * WAD); // Simulates 10 WAD of interest accumulated
```

Comment:
I'm noticing that sometimes we keep the old way of adding to address(this) and the transferring to the pool to make sure to not override existing balance but in others we set directly in the pool. I know for this PoolMock it would be the same as depositing doesn't add testGem anymore so we start with balance 0. However I'd believe we should standardize. Probably the less error prone is setBalance to the pool directly but adding up to the current balance.

Reply:
Good point. I added a standard `accrueInterest` function to the pool.